Modix/Services/Cat/CatService.cs

```diff
@@ -55,10 +74,19 @@ public struct URL
 
             do
             {
-                // Download a JSON string from the API
-                json = await DownloadCatJson(token);
+                try
+                {
+                    // Download a JSON string from the API
+                    json = await DownloadCatJson(token);
+                }
+                catch (TaskCanceledException)
+                {
+                    Log.Warning("Could not retrieve JSON string within the alloted time. Is the API Down?");
+                    return "Cat not downloaded in time";
+                }
 
-                // Check and make sure the string isn't empty before attempting to deserialize cat
+                // Check and make sure the string isn't empty before attempting to deserialize the
+                // json. If the website returns a blank string, something is wrong. Break the loop
                 if (string.IsNullOrWhiteSpace(json)) break;
 
                 //Deserialize the json retrieved from the website
```

Comment:
I'm seeing doubles (this code is identical to the code above, and can probably be extracted to a common method